xds/internal/xdsclient/transport/transport.go

```diff
@@ -268,6 +292,10 @@ func (t *Transport) sendAggregatedDiscoveryServiceRequest(stream adsStream, reso
 	if err := stream.Send(req); err != nil {
 		return fmt.Errorf("sending ADS request %s failed: %v", pretty.ToJSON(req), err)
 	}
+	if err := t.onSendHandler(&ResourceSendInfo{URL: resourceURL, ResourceNames: resourceNames}); err != nil {
```

Comment:
I don't think we need to return an error from the `onSendHandler()` callback. (In my initial comment asking to handle the resource-type not found in the map, I had asked for a log entry and an early return, and I still feel that would be sufficient).

If we return an error from the callback and return that same error from this function, it would result in the stream being closed and recreated. See the `send()` method.

Reply:
Oh from your last comment i gathered I had to return the err. Let me update that. Thanks for catching that!